R/load_pkgs.R

```diff
@@ -134,12 +161,26 @@ load_pkgs <- function(branch_name = NULL, git_pull = FALSE, from_commit = list()
         if (!check_remote) {
           cat("\033[0;96mRemote URL exists...\033[0m\n")
           cat("\033[0;96mPerforming git pull...\033[0m\n")
-          system2(
+          pull_error <- system2(
             "cd",
             args = c(path, "&& git pull"),
             stdout = FALSE,
             stderr = FALSE
           )
+          if (pull_error) {
+            message <- paste(
+              "\033[0;91m`git pull` failed.",
+              "Please check your local changes for\033[0;94m",
+              pkg, "\033[0m\n"
+            )
+            if (load_verbose == "error") {
+              stop(message)
+            } else if (load_verbose == "warning") {
+              warning(message)
+            } else if (load_verbose != "silent") {
+              cat(message)
+            }
+          }
```

Comment:
We can do the same here with the `git checkout error`:
* use switch statement
* make the message more user friendly

Reply:
Agreed, made the changes!
While we're at #45 let's also remember to create a nice utils for this `log_verbose` logic that we can use in multiple places.